apis/discoveryengine/v1alpha1/targetsite_reference.go

```diff
@@ -0,0 +1,155 @@
+// holds the GCP identifier for the KRM object.
+type TargetSiteRef struct {
+	// A reference to an externally managed DiscoveryEngineDataStoreTargetSite resource.
+	// Should be in the format "projects/<projectID>/locations/<location>/targetsites/<targetsiteID>".
+	External string `json:"external,omitempty"`
+
+	// The name of a DiscoveryEngineDataStoreTargetSite resource.
+	Name string `json:"name,omitempty"`
+
+	// The namespace of a DiscoveryEngineDataStoreTargetSite resource.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// TargetSiteLink defines the full identity for a DataStoreTargetSite
+//
+// +k8s:deepcopy-gen=false
+type TargetSiteLink struct {
+	*DiscoveryEngineDataStoreID
+	TargetSite string
+}
+
+func (l *TargetSiteLink) String() string {
+	return l.DiscoveryEngineDataStoreID.String() + "/siteSearchEngine/targetSites/" + l.TargetSite
+}
+
+// NormalizedExternal provision the "External" value for other resource that depends on DiscoveryEngineDataStoreTargetSite.
+// If the "External" is given in the other resource's spec.DiscoveryEngineDataStoreTargetSiteRef, the given value will be used.
+// Otherwise, the "Name" and "Namespace" will be used to query the actual DiscoveryEngineDataStoreTargetSite object from the cluster.
+func (r *TargetSiteRef) NormalizedExternal(ctx context.Context, reader client.Reader, otherNamespace string) (string, error) {
+	if r.External != "" && r.Name != "" {
+		return "", fmt.Errorf("cannot specify both name and external on %s reference", DiscoveryEngineDataStoreTargetSiteGVK.Kind)
+	}
+	// From given External
+	if r.External != "" {
+		if _, err := ParseTargetSiteExternal(r.External); err != nil {
+			return "", err
+		}
+		return r.External, nil
+	}
+
+	// From the Config Connector object
+	if r.Namespace == "" {
+		r.Namespace = otherNamespace
+	}
+	key := types.NamespacedName{Name: r.Name, Namespace: r.Namespace}
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(DiscoveryEngineDataStoreTargetSiteGVK)
+	if err := reader.Get(ctx, key, u); err != nil {
+		if apierrors.IsNotFound(err) {
+			return "", k8s.NewReferenceNotFoundError(u.GroupVersionKind(), key)
+		}
+		return "", fmt.Errorf("reading referenced %s %s: %w", DiscoveryEngineDataStoreTargetSiteGVK, key, err)
+	}
+	// Get external from status.externalRef. This is the most trustworthy place.
+	actualExternalRef, _, err := unstructured.NestedString(u.Object, "status", "externalRef")
+	if err != nil {
+		return "", fmt.Errorf("reading status.externalRef: %w", err)
+	}
+	if actualExternalRef == "" {
+		return "", k8s.NewReferenceNotReadyError(u.GroupVersionKind(), key)
+	}
+	r.External = actualExternalRef
+	return r.External, nil
+}
+
+// NewTargetSiteLinkFromObject builds a TargetSiteLink from the Config Connector object.
+func NewTargetSiteLinkFromObject(ctx context.Context, reader client.Reader, obj *DiscoveryEngineDataStoreTargetSite) (*DiscoveryEngineDataStoreID, *TargetSiteLink, error) {
```

Comment:
Just curious, this is not called and also not from the SciFi template. Is it from Gemini?